src/webui/api/torrentscontroller.cpp

```diff
@@ -1622,3 +1792,246 @@ void TorrentsController::setSSLParametersAction()
 
     torrent->setSSLParameters(sslParams);
 }
+
+void TorrentsController::fetchMetadataAction()
```

Comment:
Too long function with too many `return` statements. Poorly readable.
At a minimum, I would insist you to reformat the `return` statements to `else` branches.